src/components/Request/index.js

```diff
@@ -0,0 +1,52 @@
+import React, {Component} from 'react'
+import {startsWith} from 'lodash'
+import {logout} from 'utils/authentication'
+import {default as RequestBase} from './components/Request'
+
+const statusCodes = {
+  unauthorized: 401,
+}
+
+export default class Request extends Component {
+
+  getUrlWithBase = () => `${process.env.REACT_APP_EGGHEAD_BASE_URL}${this.props.url}`
+
+  getHeaders = () => ({
+    ...this.props.headers,
+    Authorization: localStorage.token
+      ? `Bearer ${localStorage.token}`
+      : null,
+    'Content-Type': 'application/json',
+  })
+
+  handleError = (error) => {
+    if (error.status === statusCodes.unauthorized) {
+      logout()
+    }
+    if (this.props.onError) {
+      this.props.onError(error)
+    }
+  }
+
+  render() {
+    const {url, children, ...rest} = this.props
+    return (
+      <RequestBase
+        {...rest}
+        url={startsWith(url, '/') ? this.getUrlWithBase() : url}
+        headers={this.getHeaders()}
+        onError={this.handleError}
+      >
+        {({request, running, error, data, response}) => {
+          return children({
+            request,
+            running,
+            error,
+            data,
+            response,
+          })
+        }}
+      </RequestBase>
```

Comment:
I ❤️  render callbacks (aka Function As Child) components. So much power with so little code. Here we add default error handling with endpoint status (logout if forbidden, token is expired), add default loading/error component rendering etc.